libs/wingsdk/src/cloud/bucket.ts

```diff
@@ -55,6 +58,159 @@ export abstract class Bucket extends Resource {
    * referencing a file from the local filesystem.
    */
   public abstract addObject(key: string, body: string): void;
+
+  /**
+   * Iterates over the topics and supply their sim handler
+   * @returns an object of Bucket event types (keys) and their topic handlers (values)
+   */
+  protected convertTopicsToHandles() {
+    const topicMap: Record<string, string> = {};
+
+    this.topics.forEach((value, key) => {
+      topicMap[key] = `\${${value.node.path}#attrs.handle}`;
+    });
+
+    return topicMap;
+  }
+
+  /**
+   * creates a topic for subscribing notification event
+   * @param actionType
+   * @returns the created topi
+   */
+  protected createTopic(actionType: BucketEventType): Topic {
+    const hash = this.node.addr.slice(-8);
+
+    const topic = Topic._newTopic(
+      this.node.scope!, // ok since we're not a tree root
```

Comment:
Also I think if we do this, then we can simplify the `id` of the topic in the line below since we no longer need to name child resources defensively to prevent naming conflicts:
```js
`on_${actionType.toLowerCase()}`
```